docs/specs/code/TRANSACTION_MEMO.md

```diff
@@ -0,0 +1,239 @@
+* [Upload single file](https://dropbox.github.io/dropbox-sdk-js/Dropbox.html#filesUpload__anchor): Create a new file with the contents provided in the request.
+* [Download single file](https://dropbox.github.io/dropbox-sdk-js/Dropbox.html#filesDownload__anchor): Download a file from a user's Dropbox.
+* [Search individual file](https://dropbox.github.io/dropbox-sdk-js/Dropbox.html#filesSearch__anchor): Searches for files and folders. Note: Recent changes may not immediately be reflected in search results due to a short delay in indexing.
+* [Download folder as a ZIP file](https://dropbox.github.io/dropbox-sdk-js/Dropbox.html#filesDownloadZip__anchor): Download a folder from the user's Dropbox, as a zip file. The folder must be less than 20 GB in size and have fewer than 10,000 total files. The input cannot be a single file. Any single file must be less than 4GB in size.
+
+Using Dropbox will require to [create an app](https://www.dropbox.com/developers/reference/developer-guide). The user will have to connect his/her account and grant permissions, after which an [access token](https://www.dropbox.com/developers/reference/oauth-guide) will be generated. This access token must be safely stored locally in order to make authorized requests on behalf of the user (refer to the [encryption](https://github.com/Emurgo/yoroi-frontend/blob/737595fec5a89409aacef827d356c9a1605515c0/docs/specs/code/ENCRYPT.md) spec). 
+
+Note: API rate limits will apply in a [per user/app basis](https://www.dropbox.com/developers/reference/data-ingress-guide), so it shouldn't be a concern for Yoroi itself.
+
+### Fault tolerance
+
+In the case of a failure in the external service, we must consider to keep the memos in local storage until they can be successfuly uploaded. There could be several sources of failure for a memo to be uploaded:
+
+* External storage is unavailable
+* The access token is expired
+* The user revoked access for the Yoroi app registered in the external service
+* The file already exists
+* The user exceeded the limit of API calls to the external storage
+
+Therefore, we must consider storing an error message field along with the memo when saving it to local storage. Also, notice that when a memo fails to be uploaded, another instance of the same wallet but in a different device could add a memo for the same transation. In this case, we apply the *first come, first served* principle and keep the first memo that was uploaded. 
+
+## Local storage
+
+We need to save the following items into local storage:
+
+* LS1: `memoPassword` (string): encrypted memo password. This password is used to encrypt/decrypt memo files. The memo password itself is encrypted/decrypted using the spending password (password-based strategy).
+* LS2: `dropboxAccessToken` (string): access token provided by Dropbox after the user granted access to the Yoroi app. This token is used to connect to Dropbox API on behalf of the user.
```

Comment:
I would avoid the name `dropboxAccessToken` since we may have more  services in the future. Probably it's better to have some object with a identifier representing the service name (with only Dropbox for now) and a generic `token` field.